src/main/java/it/gov/pagopa/admissibility/rest/anpr/residence/ResidenceAssessmentRestClientImpl.java

```diff
@@ -0,0 +1,135 @@
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.headRequest.addresseeCode}") String headRequestAddresseeCode,
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.headRequest.operationRequest}") String headRequestOperationRequest,
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.headRequest.sendType}") String headRequestSendType,
+
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.dataRequest.personalDetailsRequest}") String dataRequestPersonalDetailsRequest,
+
+                                             ObjectMapper objectMapper) {
+        this.agidJwtSignature = agidJwtSignature;
+
+        this.objectMapper = objectMapper;
+
+        HttpClient httpClient = anprWebClient.getHttpClientSecure();
+
+        this.webClient = WebClient.builder()
+                .clientConnector(new ReactorClientHttpConnector(httpClient))
+                .baseUrl(residenceAssessmentBaseUrl)
+                .build();
+
+
+        this.headRequestSenderCode = headRequestSenderCode;
+        this.headRequestAddresseeCode = headRequestAddresseeCode;
+        this.headRequestOperationRequest = headRequestOperationRequest;
+        this.headRequestSendType = headRequestSendType;
+        this.dataRequestPersonalDetailsRequest = dataRequestPersonalDetailsRequest;
+    }
+
+    @Override
+    public Mono<RispostaE002OKDTO> getResidenceAssessment(String accessToken, String fiscalCode) {
+        String requestDtoString = convertToJson(generateRequest(fiscalCode));
+        String digest = createDigestFromPayload(requestDtoString);
+        return webClient.post()
+                .uri("/anpr-service-e002")
+                .contentType(MediaType.APPLICATION_JSON)
+                .headers(httpHeaders -> {
+                    httpHeaders.setContentType(MediaType.APPLICATION_JSON);
+                    httpHeaders.setBearerAuth(accessToken);
+                    httpHeaders.add("Agid-JWT-Signature", agidJwtSignature.createAgidJwt(digest));
+                    httpHeaders.add("Content-Encoding", "UTF-8");
+                    httpHeaders.add("Digest", digest);
+                })
+                .bodyValue(requestDtoString)
+                .retrieve()
+                .bodyToMono(RispostaE002OKDTO.class);
+        //TODO define error code for retry
+    }
+
+    private String convertToJson(RichiestaE002DTO requestE002DTO) {
+        try {
+            return objectMapper.writeValueAsString(requestE002DTO);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Error converting request dto in JSON",e);
+        }
+    }
+
+    private String createDigestFromPayload(String request) {
```

Comment:
make it static, rename into createSHA256Digest and move it inside it.gov.pagopa.admissibility.utils.Utils